compiler/src/dotty/tools/dotc/transform/TreeMapWithStages.scala

```diff
@@ -92,6 +92,13 @@ abstract class TreeMapWithStages(@constructorOnly ictx: Context) extends TreeMap
       }
 
       tree match {
+        case Apply(Select(Quoted(quotedTree), _), _) if quotedTree.isType =>
+          dropEmptyBlocks(quotedTree) match
+            case Spliced(t) =>
+              // '[ x.$splice ] --> x
+              transform(t)
+            case _ =>
+              super.transform(tree)
```

Comment:
Does the branch catch the case where the splice is in a big type, such as `'{ e: F[$x]}`?

Reply:
Yes